layer/components/Amount/Display.vue

```diff
@@ -0,0 +1,109 @@
+<script setup lang="ts">
+import { BigNumberInBase } from "@injectivelabs/utils";
+
+const props = withDefaults(
+  defineProps<{
+    amount: string | number | BigNumberInBase;
+    decimals?: number;
+    useSubscript?: boolean;
+    useAbbreviation?: boolean;
+    abbreviationThreshold?: number;
+    noTrailingZeros?: boolean;
+    subscriptThresholdDecimals?: number;
+    subscriptDecimals?: number;
+  }>(),
+  {
+    decimals: 6,
+    useSubscript: false,
+    useAbbreviation: false,
+    abbreviationThreshold: 1_000_000,
+    noTrailingZeros: false,
+    subscriptThresholdDecimals: 4,
+    subscriptDecimals: 4,
+  },
+);
```

Comment:
can sort by line length + for boolean optional props, no need for false default value